### PR TITLE
remove duplicate annotation in nats statefulset

### PR DESCRIPTION
**Pull request by sudarshan-uc**
combined config from two metadata.annotations

## Description

There are two annotation sections in the NATS statefulsets, 

## Related Issues

fixes: https://github.com/astronomer/issues/issues/6284

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

Do not merge this PR until it lists which release branches this PR should be merged / cherry-picked into.

### CODE REVIEW

File: charts/nats/templates/statefulset.yaml [hunk 1]
```diff
@@ -22,8 +22,11 @@ spec:
   serviceName: {{ template "nats.name" . }}
   template:
     metadata:
-      {{- if or .Values.podAnnotations .Values.exporter.enabled }}
       annotations:
+        checksum/nats-config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+      {{- if and  .Values.global.nats.jetStream.enabled .Values.global.nats.jetStream.tls }}
+        checksum/nats-tls: {{ include (print $.Template.BasePath "/nats-jetstream-tls-secret.yaml") . | sha256sum }}
+      {{- end }}
```
Comment (pgvishnuram) [inline charts/nats/templates/statefulset.yaml:29]: @danielhoherd this is a new change on top of sudharsan change

**Review by rishkarajgi — APPROVED** [review]

**Comment by rishkarajgi** [comment]
Tests are failing

**Comment by pgvishnuram** [comment]
Let me re-run them , adding basic test case for annotation validation

**Comment by rob-1126** [comment]
Can we explicitly test the upgrade path here I am expecting it to result in failures and that variable being completely unset for a period of time between the two due to helm issue 10741. What would the impact to the platform be during that time?

**Comment by pgvishnuram** [comment]
i dont think this will cause an issue for the customers

**Pull request merged**